File: python/digito_verificador.py
def generar_digito_verificador(ruc, base_max=11):
    try:
        ruc = ruc if isinstance(ruc, str) else str(ruc)
        suma = 0
        for index, ruc_index in enumerate(range(len(ruc) - 1, -1, -1)):
            suma += int(ruc[ruc_index]) * (index + 2)
        r = suma % base_max

        return str(base_max - r if r > 1 else 0)
    except:
        return ''

File: javascript/digitoVerificador.js
digitoVerificador = function (ruc, baseMax = 11) {
    ruc = typeof ruc === 'string' ? ruc : ruc.toString()
    var resultado = 0
    var index = 0;
    for (var rucIndex = ruc.length - 1; rucIndex >= 0; rucIndex--) {
        resultado += parseInt(ruc[rucIndex]) * (index + 2);
        var r = resultado % baseMax
        index++;
    };
    var verificador = r > 1 ? baseMax - r : 0
    return verificador
}
